### DISCUSSION TITLE
How to specify a dict config

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8545] [Q&A]
How do you specify the following schema configuration on an op():

config:
    dict_config: { 'key1': { 'subkey1': 'value1', 'subkey2': 'value2' },
                           'key2': { 'subkey1': 'value1', 'subkey2': 'value2' } }

where key1 and key2 are user specific and not known.  Looks like Selector does this but only allows one entry to be configured in yaml file.  Looks like I can just call out a dict() but then the subdict is not validated, right?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1647954673784519?thread_ts=1647954673.784519&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can use `Map` to define a config dict:
```python
from dagster import Map

@op(config_schema={"dict_config ": Field(Map({str: int}})))
def my_op(context):
    print(context.op_config.items())
```

API doc: https://docs.dagster.io/_apidocs/config#dagster.Map
